### PR TITLE
Refactoring assert raises in python algorithm tests part 5

### PR DESCRIPTION
**Pull request by jhaigh0**
**Description of work.**

Fifth (and last) part of the work to remove the blanket use of `assertRaises` when testing algorithm input validation in favour of `assertRaisesRegex`. This way we can expect a certain exception rather than accepting any possible `RuntimeException` which could be an unrelated bug (there were a few found during in the tests from this pr).

Main issue is #34570

**To test:**

 - All tests should pass (handled by Jenkins)
 - Code review

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/TotScatCalculateSelfScattering.py [hunk 1]
```diff
@@ -91,8 +91,11 @@ def PyExec(self):
 
         placzek_kwargs = {"InputWorkspace": raw_ws, "IncidentSpectra": fit_spectra, "ScalebyPackingFraction": False, "Order": placzek_order}
         sample_temp = self.getPropertyValue("SampleTemp")
-        if placzek_order == 2 and sample_temp:
-            placzek_kwargs.update({"SampleTemperature": sample_temp})
+        if placzek_order == 2:
+            if sample_temp:
+                placzek_kwargs.update({"SampleTemperature": sample_temp})
+            else:
+                raise ValueError("SampleTemperature must be provided")
```
Comment (robertapplin) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/TotScatCalculateSelfScattering.py:98]:
```suggestion
        if placzek_order == 2:
            if not sample_temp:
                raise ValueError("SampleTemperature must be provided")
            placzek_kwargs.update({"SampleTemperature": sample_temp})
```

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/MSDFitTest.py [hunk 2]
```diff
@@ -73,36 +73,76 @@ def test_fail_spec_min(self):
...
     def test_fail_spec_range(self):
         """
-        Tests validation for SpecMax >= SpecMin.
+        Tests validation for SpecMin >= SpecMax.
         """
 
-        self.assertRaises(RuntimeError, MSDFit, InputWorkspace=self._ws, XStart=0.0, XEnd=5.0, SpecMin=1, SpecMax=0, OutputWorkspace="msd")
+        self.assertRaisesRegex(
+            RuntimeError,
+            "SpecMin must be less then SpecMax",
```
Comment (robertapplin) [inline Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/MSDFitTest.py:103]:
```suggestion
            "SpecMin must be less than SpecMax",
```
I think there is a typo in this exception

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/MSDFitTest.py [hunk 3]
```diff
@@ -73,36 +73,76 @@ def test_fail_spec_min(self):
...
     def test_fail_x_range(self):
         """
-        Tests validation for XStart < XEnd.
+        Tests validation for XStart > XEnd.
         """
 
-        self.assertRaises(RuntimeError, MSDFit, InputWorkspace=self._ws, XStart=10.0, XEnd=5.0, SpecMin=0, SpecMax=0, OutputWorkspace="msd")
+        self.assertRaisesRegex(
+            RuntimeError,
+            "XStart must be less then XEnd",
```
Comment (robertapplin) [inline Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/MSDFitTest.py:120]:
```suggestion
            "XStart must be less than XEnd",
```
same here

**Review by robertapplin — CHANGES_REQUESTED** [review]
The code looks good. I had one minor code style suggestion, and spotted a couple of typos in some of the exceptions

**Comment by jhaigh0** [comment]
> The code looks good. I had one minor code style suggestion, and spotted a couple of typos in some of the exceptions

Will need to change those strings where they are raised too.

**Review by robertapplin — APPROVED** [review]
Thanks for the changes, looks good 👍

**Pull request merged**